src/libraries/Microsoft.Extensions.Logging/ref/Microsoft.Extensions.Logging.cs

```diff
@@ -85,6 +85,15 @@ public partial class LoggerFilterRule
         public string ProviderName { get { throw null; } }
         public override string ToString() { throw null; }
     }
+    [System.AttributeUsageAttribute(System.AttributeTargets.Method)]
+    public sealed partial class LoggerMessageAttribute : System.Attribute
```

Comment:
I know this is still up in the air - but I assumed we weren't putting the attribute in the library's `ref` assembly unless we could ensure users get a compile-time error if the source generator doesn't run.

Reply:
I think we need to file an issue on roslyn to help solve this.  I think this generator is a great case study to drive such an issue.  Especially given that in Preview4 the reference assembly will be in the ASP.NETCore shared framework but the generator will only be in the NuGet package.

Reply:
> unless we could ensure users get a compile-time error if the source generator doesn't run.

A way to do this would be to provide an analyzer that validates if the corresponding partial-implementation part is there, and errors if it is not.  If that's a requirement of this attribute, it should be responsible for delivering the analyzer that validates that piece IMO.

Reply:
I was just about to reply with your message here @CyrusNajmabadi :) you beat me to it.

Reply:
> A way to do this would be to provide an analyzer that validates if the corresponding partial-implementation part is there, and errors if it is not. If that's a requirement of this attribute, it should be responsible for delivering the analyzer that validates that piece IMO.

What if the analyzer isn't passed to the compiler?  That's just the same problem.  I don't think adding redundancy in another optional component is a solution.

Reply:
cc: https://github.com/dotnet/roslyn/issues/52749#issuecomment-823499820